src/engine.rs

```diff
@@ -8,6 +8,7 @@ use content_length::ContentLength;
 pub struct Engine {
     urls: Vec<String>,
     method: Method,
+    headers: Vec<String>,
```

Comment:
I think it's better if this takes type of `Vec<(String, String)>` and we do the processing of the inputs in `main.rs`